### ISSUE TITLE
Custom ticks of XAxis fixed

### ISSUE THREAD

**Issue by tuanddd** [recharts/recharts#1028]
## Do you want to request a feature or report a bug?
It's more of an implementation problem
## What is the current behavior?
I tried `<XAxis type="number" ticks={['09:00', '10:00', '11:00']} />` but both graph and the xaxis disappear
## What is the expected behavior?
I want the XAxis to display fixed ticks like different hour in a day, and each coming new data will represent a seconds of it (much like of an stock chart)
## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
I am using `"recharts": "^1.0.0-alpha.6"`

**Comment by huei-tan-sociomantic** [comment]
well, it works perfectly fine in this example http://jsfiddle.net/jh461xmL/2/ with the version alpha.6.

**Comment by TuanDSE62171** [comment]
Oh yeah but I forgot to mention though that the chart I'm using is going to be receiving data in realtime. And if I recall correctly, the XAxis in the above example will not be "fixed" (e.g "Page G" is going to be pushed back as new data is added).
I did think of using d3 time scale solution but it's the last resort as I have to rewrite the chart component 
But anyways, amazing library! Also thank you for showing the example :)

**Comment by anze555** [comment]
I am having a similar issue. I would like to draw ticks (and grid) at whole hours (`10:00`, `11:00`, `12:00`,...), but the data I have is scattered at different times. I thought I could have numerical X values (timestamp) and use `ticksFormatter` to display it as hours, but it turns out ticks are only displayed if the supplied data coincides with the X values (that is, if I actually have a value at `10:00`).

In other words: in the example you posted `Page C` actually exists in `data` - if you remove it, both `Page A` (not sure why that is?) and `Page C` go missing.

Is there a way I can place ticks on `XAxis` independently of the data?

**Comment by anze555** [comment]
@TuanDSE62171 I read some other tickets and found a way to do this. The idea is to treat time as `number` (as you have specified in XAxis's type) and to explicitly supply  `domain`, `ticks` and `tickFormatter`. Working example:

```
import moment from 'moment';  // `yarn add moment`
const data = [
  {t: 1080001800, uv: 4000, pv: 2430, amt: [2400,2500]},
  {t: 1080005400, uv: 3000, pv: 1398, amt: [1210,1600]},
  {t: 1080009000, uv: 2000, pv: 2340, amt: [2290,2550]},
  {t: 1080012600, uv: 2780, pv: 2108, amt: [2000,2200]},
  {t: 1080016200, uv: 1890, pv: 2200, amt: [2181,2300]},
  {t: 1080019800, uv: 2390, pv: 2600, amt: [2500,2700]},
  {t: 1080023400, uv: 3490, pv: 2120, amt: [2100,2150]},
];
```

Then define `XAxis` like this:
```
          <XAxis 
            dataKey="t" 
            type="number" 
            domain={[1080000000, 1080025200]}
            tickFormatter={(tick) => moment(tick * 1000).format('HH:mm')}
            ticks={[1080000000, 1080003600, 1080007200, 1080010800, 1080014400, 1080018000, 1080021600, 1080025200]}
          />
```

Using `interval` didn't work for me - I got an empty axis whatever I did. Also, adding `scale="time"` led to empty axis.

As far as I'm concerned, my issue is solved. Hope it helps.

**Comment by mrt123** [comment]
@anze555 how did you get tooltip to display formatted date instead of timestamp?

**Comment by anze555** [comment]
This is what `tickFormatter` does in the example I posted... hope it helps.

**Comment by maracuja-juice** [comment]
@anze555 Do you know of some method to calculate these values?

**Comment by anze555** [comment]
@maracuja-juice If you mean something like a library, then unfortunately no, but I would be interested if you ever find it. :)

For my own purposes I made a function that (depending on scale, time range and width of chart in pixels) returns the appropriate ticks. What is appropriate is of course a matter of taste... I wrote quite a lot of tests first, then the implementation was pretty straightforward. Not a lot of work in the end, a few hours max.

**Comment by maracuja-juice** [comment]
@anze555 I want a tick for every 10 minutes of data. Gave this a try today but I didn't achieve what I want.
(Example: possible labels are: 20:01, 20:02, 20:03, 20:04, 20:05, 20:06, 20:07, 20:08, 20:09, 20:10. Ticks should be at: 20:00 and 20:10)

**Comment by anze555** [comment]
@maracuja-juice If you only want to have ticks at 10 min, then the solution is simple... First of all, convert the dates to timestamps. Then start a loop at `Math.ceil(timeFrom / 60*10) * 60*10`, increment by `60*10` (10 minutes) and stop after you reach `Math.floor(timeTo / 60*10) * 60*10. This will get you the timestamps in range, and you can convert them to strings as indicated in above comments (using `moment(t).format(...)`). Hope it helps.

On a side note however, it might be better to post a StackOverflow question with this problem... You will get better help there.

**Comment by veddermatic** [comment]
@anze555 I came up with you solution to getting a time axis rendering on the hours, but it's dropping ticks for some reason. There are 24 timestamps in my `xTicks` array, but some go missing when the chat renders, despite there being plenty of space for them. Anyone seen this before?
![Screen Shot 2019-06-28 at 1 45 06 PM](https://user-images.githubusercontent.com/322479/60361578-984b3100-99ac-11e9-8211-2840d77c941e.png)

**Comment by dreamsparkx** [comment]
@veddermatic This same thing is happening to me. Did you find a solution?

**Comment by Slebluue** [comment]
@dreamsparkx @veddermatic did you guys try setting the interval to 0?

http://recharts.org/en-US/api/XAxis#interval
